platform/broadcom/sonic-platform-modules-accton/as9716-32d/sonic_platform/pcie.py

```diff
@@ -17,3 +17,41 @@ class Pcie(PcieUtil):
 
     def __init__(self, platform_path):
         PcieUtil.__init__(self, platform_path)
+        self._conf_rev = self.__get_conf_rev()
+
+    def __get_conf_rev(self):
+        """
+        Retrieves the system EEPROM label revision and selects the corresponding pcie.yaml file.
+
+        This function initializes the PDDF API and loads the EEPROM data. It retrieves the
+        TLV field for the label revision from the EEPROM. If the label is valid, it decodes
+        the ASCII value and attempts to find a matching pcie.yaml file based on the revision.
+
+        Returns:
+            str: The label revision if a matching pcie.yaml file is found.
+            None: If the label revision is not found or any error occurs.
+        """
+        try:
+            import os
+            from sonic_platform_pddf_base import pddfapi
+            from sonic_platform.eeprom import Eeprom
+
+            pddf_obj = pddfapi.PddfApi()
+            eeprom = Eeprom(pddf_obj, "{}") # The content of pd-plugin.json is not needed.
+            if eeprom is not None:
+                 # Try to get the TLV field for the label revision
+                (is_valid, results) = eeprom.get_tlv_field(eeprom.eeprom_data, eeprom._TLV_CODE_LABEL_REVISION)
+                if not is_valid or results[2] is None:
+                    return None
+
+                label_rev = results[2].decode('ascii')
+
+                for rev in (label_rev, label_rev[:-1]):
```

Comment:
For the label_rev, it may be R0BA, R01B
In you implementation, R0BA may search R0BA first, and then R0B. Therefore, pcie_R0BA.yaml is used.
In you implementation, R01B may search R01B first, and then R01. Therefore, pcie_R01.yaml is used.